src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.Extension/syntaxes/aspnetcorerazor.tmLanguage.yml

```diff
@@ -27,6 +27,55 @@ injections:
         end: '(\})'
         endCaptures:
           1: { name: 'punctuation.curlybrace.close.cs' }
+  # 'L:meta.attribute':
+  #   patterns:
+  #     - name: 'meta.attribute.razor.explicit'
+  #       begin: '(")(@)\('
+  #       beginCaptures:
+  #         0: { name: 'keyword.control.cshtml' }
+  #         1: { patterns: [ include: '#transition' ] }
+  #       # begin: '"(@)\('
+  #       # beginCaptures:
+  #       #   0: { name: 'keyword.control.cshtml' }
+  #       #   1: { patterns: [ include: '#transition' ] }
+  #       patterns:
+  #         - include: 'source.cs#expression'
+  #       end: '\)'
+  #       endCaptures:
+  #         0: { name: 'keyword.control.cshtml' }
+  #     - name: 'meta.attribute.razor.implicit'
+  #       begin: '("[^@]*)(@)'
+  #       beginCaptures:
+  #         0: { name: 'keyword.control.cshtml' }
+  #         1: { patterns: [ include: '#transition' ] }
+  #       patterns:
+  #         - include: '#await-prefix'
+  #         - include: '#implicit-expression-body'
+  #       end: '(?=[\s<>\{\}\)\]])(?=([^"]*")'
+
+
+  'L:meta.attribute string.quoted.double.html':
+    patterns:
+      - name: 'meta.attribute.razor.explicit'
+        begin: '@\('
+        beginCaptures:
+          0: { name: 'keyword.control.cshtml' }
+          1: { patterns: [ include: '#transition' ] }
+        patterns:
+          - include: 'source.cs#expression'
+        end: '\)'
+        endCaptures:
+          0: { name: 'keyword.control.cshtml' }
+      - name: 'meta.attribute.razor.implicit'
+        begin: '@'
+        beginCaptures:
+          0: { name: 'keyword.control.cshtml' }
+          1: { patterns: [ include: '#transition' ] }
+        patterns:
+          - include: '#await-prefix'
+          - include: '#attribute-implicit-expression-body'
```

Comment:
Added new `attribute-implicit-expression-body` which is essentially an extension of `implicit-expression-body` but with `"` recognized as an `end` character (to ensure the C# context is correct escaped when at the end of an attribute value (ex. `<button someattribute="@DateTime|" />` where `|` is the cursor)

Reply:
You can expand the `implicit-expression-body` rules to have that. Honestly we should probably have that in them anyways

Reply:
Ya I think updating the implicit-expression-body rules to have `"` and then nuking the attribute variants will fix most of your coloring issues. Right now you're missing some of the expression joiners like dots which is why some things are getting colored real funky.

Reply:
Okay cool, I was hesitant in modifying the existing rules in case there were legacy reasons for it not to be included.

Reply:
This worked beautifully, resolved the issue of the `Now` of `DateTime.Now` not being colorized correctly. 🎉  Thanks!!

![image](https://user-images.githubusercontent.com/14852843/80156474-db4c4580-8578-11ea-919d-b05bcf3c3b9f.png)

Reply:
I'd also add a single quote to this list as well

Reply:
Yep, updating the rules + tests to accommodate  single quotes.

Reply:
I'm surprised the `.`'s and parens are being colored they way they are. We may need to address that. I think the reason is that they don't technically have any category so by default they're opting into whatever their parent scope is.

Reply:
They seem to be in the correct scope, just missing some colorization rules. Could we just add colorization for them instead of changing the grammar?

![image](https://user-images.githubusercontent.com/14852843/80156918-e784d280-8579-11ea-9d9e-144f3e02e4cd.png)


![image](https://user-images.githubusercontent.com/14852843/80156927-f10e3a80-8579-11ea-9071-f9f3155c0623.png)

Reply:
@ryanbrandenburg when you played around with VSCode extensions providing themed colors (for TagHelpers for instance) are there any drawbacks/limitations? Or would we be able to provide scopes for things like punctuation.parenthesis.close.cs no problem?

@TanayParikh We can easily add this color in VS since the entire Razor theme ships in the extension: https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.RazorExtension/EmbeddedGrammars/aspnetcorerazor.tmLanguage.tmTheme

Reply:
As I recall semantic colorization wasn't able to provide colors based on scope, only on tokenType. But we've had a pandemic since then so I couldn't swear to it.

Reply:
Ahh ok, that makes more sense to me. I didn't think it was possible to provide colors in that way either. @TanayParikh I'd focus on ensuring the VS colorization is correct. In this case, when you get there let me know and I can show you how to get the appropriate "VS Classification" for colors

Reply:
The latest commit features VS colorization. https://github.com/dotnet/aspnetcore-tooling/pull/1824#issuecomment-619260011